### ISSUE TITLE
Reliable way of getting video and audio containing formats from JSON

### ISSUE THREAD

**Issue by christoph-heinrich** [yt-dlp/yt-dlp#4373]
### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.06.29** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I am currently working on the [ytdl_hook](https://github.com/mpv-player/mpv/blob/master/player/lua/ytdl_hook.lua) from mpv.
It runs `yt-dlp` with `-J` and then parses the JSON.
Currently it just assumes that everything that doesn't have a `vcodec` set (`== null` or `== 'none'`) is an audio format. The problem with that is that things like storyboards from YouTube also get recognized as audio formats.

The README says:

> ba, bestaudio: Select the best quality audio-only format.  Equivalent to best*[vcodec=none]

 Indicating that logic to be true.
But the README also says:

>  ba*,  bestaudio*:  Select  the  best  quality  format  that  contains  audio.  It may also contain video.  Equivalent to best*[acodec!=none]

Implying that all formats that have audio are guaranteed to have `acodec != null` and `acodec != 'none'`.

The README however never states that `!= null` and `!= none` are guaranteed for vcodec or acodec if the format contains video or audio respectively.

Is there a reliable way of getting all the formats containing video and all the formats containing audio without also getting any other formats (e.g. storyboards from YouTube) from the JSON? Also it would be good to know if you know that solution to also work for youtube-dl.

P.S. I'm not a native english speaker, so maybe I'm wrong on this.
The README contains 

> b*, best*: Select the best quality format that contains either a video or an audio

but I think the word 'either' indicates xor, where as this should be a logical or. I suggest changing it to 'video and/or audio' to make it less ambiguous.

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by pukkandan** [comment]
if `acodec!='none'`, the format has an audio. Similar for video

Eg:
video-only = `vcodec!='none' && acodec=='none'`
audio-only = `vcodec=='none' && acodec!='none'`
pre-merged = `vcodec!='none' && acodec!='none'`
storyboard = `vcodec=='none' && acodec=='none'`

PS: `null` represents that we don't know what the codec is, and is not the same as `'none'`

---

Don't close the issue. I'll take a look at your suggestions about the README later

**Comment by christoph-heinrich** [comment]
Wow that was fast, thank you.

**Comment by christoph-heinrich** [comment]
I just tested it, and unfortunately this still recognizes non video formats from [this json](https://github.com/christoph-heinrich/mpv-quality-menu/issues/3#issuecomment-1185575186) as videos. But I guess that just means that yt-dlp just doesn't know that they are not videos.

As a solution to that, I ended up also checking `resolution != null` and `resolution != "audio only"`, but I guess that wouldn't be a good general solution then.
